### ISSUE TITLE
Job Selection Tools

### ISSUE THREAD

**Issue by SneakyTactician** [SneakyTactician/MagicalLife#110]
Clicking on one command type doesn't unselected the others

**Comment by ABuonanni** [comment]
For what I can see this behaviour isn't implement at all. So this is not a bug to find and fix but a feature to implement, right?

**Comment by SneakyTactician** [comment]
Yeah, not sure why I marked this as a bug. Must've been late or something.

**Comment by ABuonanni** [comment]
I'm looking at this issued and I was thinking that action buttons (Chop, Mine and TillDirt) should have a public boolean property called like "IsSelected" or "HasSelectedAspect" that the InGameGUI can use to set its buttons appeareance. The property setter code will choose the right texture for the button based on the value of InGameGUI's Selected property. Now this behaviour should be applied to all three buttons and I were wondering wheter was the case to create a new class (called something like ActionButton, to hold all common action buttons behaviour) to implements this in or to implement it in the MonoButton class (but I don't like this second option as much as the first). What do you say?

**Comment by SneakyTactician** [comment]
I like the first option, as not all buttons are toggle on/toggle off. However, I would ask you to consider making a class that extends MonoButton that implements that behavior. Then make some sort of manager class that handles only having one togglable button active at a time. What do you think of that?

Thomas

________________________________

**Comment by ABuonanni** [comment]
I like the idea. I'm gonna implement a class that extends MonoButton and a class, let say "ActionButtonsManager", that handles only having one togglable button active at a time.

**Comment by SneakyTactician** [comment]
I'd recommend branching from the "crafting branch" for this, as I've moved around the GUI API to it's own assembly in that branch.

Thomas

________________________________

**Comment by ABuonanni** [comment]
Ok, I'll do. Thanks.

**Comment by ABuonanni** [comment]
I'm resuming working on this issue. Is still Crafting the branch to work on?
Thanks.

**Comment by SneakyTactician** [comment]
No, it got merged into master. I recommend working off of master.

Thomas

________________________________
